contracts/OperatorWhitelist.sol

```diff
@@ -0,0 +1,122 @@
+    mapping(uint256 => mapping(address => bool)) public mapServiceIdOperators;
+
+    /// @dev Contract constructor.
+    /// @param _serviceRegistry Service Registry contract address.
+    constructor (address _serviceRegistry) {
+        // Check for the zero address
+        if (_serviceRegistry == address(0)) {
+            revert ZeroAddress();
+        }
+
+        serviceRegistry = _serviceRegistry;
+    }
+
+    /// @dev Controls the necessity of checking operator whitelisting statuses.
+    /// @param status True if the whitelisting check is needed.
+    function setOperatorsCheck(uint256 serviceId, bool status) external {
+        // Check that the service owner is the msg.sender
+        address serviceOwner = IToken(serviceRegistry).ownerOf(serviceId);
+        if (serviceOwner != msg.sender) {
+            revert OwnerOnly(serviceOwner, msg.sender);
+        }
+
+        // Set the operator address check requirement
+        mapServiceIdOperatorsCheck[serviceId] = status;
+    }
+    
+    /// @dev Controls operators whitelisting statuses.
+    /// @notice Operator is considered whitelisted if its status is set to true.
+    /// @param serviceId Service Id.
+    /// @param operators Set of operator addresses.
+    /// @param statuses Set of whitelisting statuses.
+    function setOperatorsStatuses(uint256 serviceId, address[] memory operators, bool[] memory statuses) external {
+        // Check for the array length
+        if (operators.length != statuses.length) {
+            revert WrongArrayLength(operators.length, statuses.length);
+        }
+
+        // Check that the arrays are not empty
+        if (operators.length == 0) {
+            revert WrongArrayLength(0, 1);
+        }
+
+        // Check that the service owner is the msg.sender
+        address serviceOwner = IToken(serviceRegistry).ownerOf(serviceId);
+        if (serviceOwner != msg.sender) {
+            revert OwnerOnly(serviceOwner, msg.sender);
+        }
+
+        bool atLeastOneOperatorWhitelisted;
+        // Set operators whitelisting status
+        for (uint256 i = 0; i < operators.length; ++i) {
+            // Check for the zero address
+            if (operators[i] == address(0)) {
+                revert ZeroAddress();
+            }
+            // Set the operator whitelisting status
+            mapServiceIdOperators[serviceId][operators[i]] = statuses[i];
+
+            // Check if at least one operator is whitelisted
+            if (statuses[i]) {
+                atLeastOneOperatorWhitelisted = true;
+            }
+        }
+        emit OperatorsWhitelistUpdated(msg.sender, serviceId, operators, statuses);
+
+        // Set the operator whitelisting check, if at least one of the operators are whitelisted
+        if (atLeastOneOperatorWhitelisted) {
+            mapServiceIdOperatorsCheck[serviceId] = true;
+            emit OperatorsWhitelistCheckSet(msg.sender, serviceId);
+        }
```

Comment:
This is confusing and will be taken out in the next PR, as the following situation is possible: whitelist operators, then de-whitelist all of them. But since the check flag was enforced and is still true, the whitelisting status will return false.